src/WidgetBase.ts

```diff
@@ -335,20 +335,24 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 	 */
 	protected addDecorator(decoratorKey: string, value: any): void {
 		value = Array.isArray(value) ? value : [ value ];
-
-		let decoratorList = decoratorMap.get(this.constructor);
-		if (!decoratorList) {
-			decoratorList = new Map<string, any[]>();
-			decoratorMap.set(this.constructor, decoratorList);
+		if (this._properties) {
```

Comment:
up to you if you want to keep this or not, but I think we can sniff if we're the instance or not with the following:
```ts
this.hasOwnProperty('constructor');
```